### PR TITLE
Add script `test_snmp_psu.py` into T0 PR checker

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

In this PR, we optimize the script `test_snmp_psu.py` to support running on kvm testbed. 
- Command `psuutil numpsus` will get the expected return code 2 on kvm testbed because of no chassis. So, on kvm testbed, we will return in advance. 
- There is no snmp psu infomation of kvm testbed. So we return in advance to skip the following checks. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
In this PR, we optimize the script `test_snmp_psu.py` to support running on kvm testbed. 
- Command `psuutil numpsus` will get the expected return code 2 on kvm testbed because of no chassis. So, on kvm testbed, we will return in advance. 
- There is no snmp psu infomation of kvm testbed. So we return in advance to skip the following checks. 

#### How did you do it?
Return in advance on kvm testbed to skip following checks.

### CODE REVIEW

**Review by xwjiang-ms — APPROVED** [review]

**Review by opcoder0 — APPROVED** [review]
LGTM

File: tests/snmp/test_snmp_psu.py [hunk 1]
```diff
@@ -21,7 +21,12 @@ def test_snmp_numpsu(duthosts, enum_supervisor_dut_hostname, localhost, creds_al
     snmp_facts = get_snmp_facts(
         localhost, host=hostip, version="v2c",
         community=creds_all_duts[duthost.hostname]["snmp_rocommunity"], wait=True)['ansible_facts']
-    res = duthost.shell("psuutil numpsus")
+    res = duthost.shell("psuutil numpsus", module_ignore_errors=True)
+
+    # For kvm testbed, we will get the expected return code 2 because of no chassis
+    if duthost.facts["asic_type"] == "vs" and res['rc'] == 2:
+        return
```
Comment (wangxin) [inline tests/snmp/test_snmp_psu.py:28]: Add some log here?

File: tests/snmp/test_snmp_psu.py [hunk 2]
```diff
@@ -40,6 +45,10 @@ def test_snmp_psu_status(duthosts, enum_supervisor_dut_hostname, localhost, cred
     psus_on = 0
     msg = "Unexpected operstatus results {} != {} for PSU {}"
 
+    # For kvm testbed, there is no snmp psu info
+    if duthost.facts["asic_type"] == "vs":
+        return
```
Comment (wangxin) [inline tests/snmp/test_snmp_psu.py:50]: Add some log here?

File: tests/snmp/test_snmp_psu.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply tests/snmp/test_snmp_psu.py:30]: Added

File: tests/snmp/test_snmp_psu.py (hunk 2 above)
Comment (yutongzhang-microsoft) [reply tests/snmp/test_snmp_psu.py:53]: Added

**Review by wangxin — APPROVED** [review]

**Pull request merged**